frontend/src/components/pages/cabins/booking/Steps/calculatePrice.ts

```diff
@@ -1,20 +1,44 @@
 import dayjs from "@/lib/date";
 
 export function calculatePrice(
-  chosenCabins: { internalPrice: number; externalPrice: number }[],
+  chosenCabins: {
+    internalPrice: number;
+    externalPrice: number;
+    internalPriceWeekend: number;
+    externalPriceWeekend: number;
+  }[],
   contactInfo: { internalParticipants: number; externalParticipants: number } | undefined,
   startDate: dayjs.Dayjs | undefined,
   endDate: dayjs.Dayjs | undefined
 ): number | undefined {
   if (!contactInfo) return NaN;
   const internalPrice = contactInfo.internalParticipants >= contactInfo.externalParticipants;
-  const pricePerNight = chosenCabins
-    .map((cabin) => (internalPrice ? cabin.internalPrice : cabin.externalPrice))
-    .reduce((sum, currentPrice) => sum + currentPrice);
-
-  if (startDate && endDate) {
-    const duration = endDate.diff(startDate, "day");
-    return pricePerNight * duration;
+  let currentDate = dayjs(startDate);
+  const finalDate = dayjs(endDate);
+  let weekdayNights = 0;
+  let weekendNights = 0;
+  let totalPrice = 0;
+  while (currentDate.isBefore(finalDate, "day")) {
+    // Last day has no night.
+    if (currentDate.day() === 5 || currentDate.day() === 6) {
+      // 5 = Friday to sat, 6 = Saturday to sunday
+      weekendNights++;
+    } else {
+      weekdayNights++;
+    }
+    currentDate = currentDate.add(1, "day");
```

Comment:
I'm slightly concerned that we have to duplicate the backend logic here. I suggest that we instead add a `price`-query or something similar and rely on the backend logic to calculate the price. That way, we won't accidentally end up with different prices.